pymc_extras/statespace/models/DFM.py

```diff
@@ -0,0 +1,388 @@
+        """
+        names = []
+
+        # Factor states
+        for i in range(self.k_factors):
+            for lag in range(self.factor_order):
+                names.append(f"factor_{i+1}_lag{lag}")
+
+        # Idiosyncratic error states
+        if self.error_order > 0:
+            for i in range(self.k_endog):
+                for lag in range(self.error_order):
+                    names.append(f"error_{i+1}_lag{lag}")
+
+        return names
+
+    @property
+    def observed_states(self) -> list[str]:
+        """
+        Returns the names of the observed states (i.e., the endogenous variables).
+        """
+        return self.endog_names
+
+    @property
+    def coords(self) -> dict[str, Sequence]:
+        coords = make_default_coords(self)
+
+        coords[FACTOR_DIM] = [f"factor_{i+1}" for i in range(self.k_factors)]
+
+        # AR parameter dimensions - add if needed
+        if self.factor_order > 0:
+            coords[AR_PARAM_DIM] = list(range(1, self.factor_order + 1))
+
+        # If error_order > 0
+        if self.error_order > 0:
+            coords["error_ar_param"] = list(range(1, self.error_order + 1))
+
+        return coords
+
+    @property
+    def shock_names(self):
+        shock_names = []
+
+        # Add names for factor shocks (one per factor)
+        for i in range(self.k_factors):
+            shock_names.append(f"factor_shock_{i+1}")
+
+        # Add names for idiosyncratic error shocks (one per observed variable)
+        if self.error_order > 0:
+            for i in range(self.k_endog):
+                shock_names.append(f"error_shock_{i+1}")
+
+        return shock_names
+
+    @property
+    def param_dims(self):
+        coord_map = {
+            "x0": (ALL_STATE_DIM,),
+            "P0": (ALL_STATE_DIM, ALL_STATE_AUX_DIM),
+            "factor_loadings": (OBS_STATE_DIM, FACTOR_DIM),
+            "factor_sigma": (FACTOR_DIM,),
+        }
+
+        if self.factor_order > 0:
+            coord_map["factor_ar"] = (FACTOR_DIM, AR_PARAM_DIM)
+
+        if self.error_order > 0:
+            coord_map["error_ar"] = (OBS_STATE_DIM, "error_ar_param")
+
+        if self.error_cov_type in ["scalar"]:
+            coord_map["error_sigma"] = ()
+        elif self.error_cov_type in ["diagonal"]:
+            coord_map["error_sigma"] = (OBS_STATE_DIM,)
+        elif self.error_cov_type in ["unstructured"]:
+            coord_map["error_sigma"] = (OBS_STATE_DIM, OBS_STATE_AUX_DIM)
+
+        return coord_map
+
+    def make_symbolic_graph(self):
+        # initial states
+        x0 = self.make_and_register_variable("x0", shape=(self.k_states,), dtype=floatX)
+
+        self.ssm["initial_state", :] = x0
+
+        # initial covariance
+        P0 = self.make_and_register_variable(
+            "P0", shape=(self.k_states, self.k_states), dtype=floatX
+        )
+
+        self.ssm["initial_state_cov", :, :] = P0
+
+        # TODO vectorize the design matrix
```

Comment:
You're going to have to double-check all of these matrix constructions if you re-ordered the states.